EIPS/eip-4672.md

```diff
@@ -0,0 +1,56 @@
+---
+eip: 4672
+title: NONCE opcode
+description: A new EVM opcode to get the current nonce of an address.
+author: 0xfoobar (@0xfoobar)
+discussions-to: https://ethereum-magicians.org/t/eip-4672-account-nonce-opcode/8171
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-02-01
+---
+
+## Abstract
+
+We propose an additional EVM opcode, `NONCE` (0x47), which could provide novel protection against sybil attacks and expand the smart contract design space. This would return `0` for an unused externally-owned address (EOA), `0` for a smart contract address, and, for example, `2` for an EOA which has sent 3 transactions. The account.nonce opcode would exactly track the nonce of the last transaction originating from that address.
+
+## Motivation
+
+NFT contracts commonly use checks on `msg.sender` to limit sybil attacks. However, this is easily circumvented by spinning up new addresses, or even nesting address creation within a smart contract. Letting smart contracts check `NONCE` would add additional costs to this, serving as a “proof of history”. While not bulletproof (as msg.sender is not), it opens up the design space to creative uses.
+
+## Specification
+
+A new opcode, `NONCE`, is introduced, with number `0x47`. The `NONCE` opcode takes one argument from the stack, and pushes to the stack the nonce corresponding to that address.
+
+The gas cost of the `NONCE` is 200.
+
+
+## Rationale
+
+The EVM makes several global variables available, examples include `block.timestamp`, `block.number`, `block.difficulty`, `msg.sender`, and `gasleft()`. It also provides several context-dependent opcodes, such as `EXTCODESIZE`, `BASEFEE`, and `BALANCE` Smart contracts can leverage these info for novel techniques, such as staking pool emissions, limiting flashloan attacks, or NFTs that embed blockchain state into their mint result.
```

Comment:
I think it is better to avoid using Solidity terms, and use EVM terms, in EIPs. This means instead of using `block.timestamp` it should read `TIMESTAMP`, etc.

However not sure a listing is needed here.